persistence-api/src/main/java/org/apache/cassandra/stargate/schema/TableMetadata.java

```diff
@@ -0,0 +1,34 @@
+package org.apache.cassandra.stargate.schema;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Table information.
+ *
+ * <p>Backend agnostic representation equivalent to {@code
+ * org.apache.cassandra.schema.TableMetadata}
+ */
+public interface TableMetadata {
+  /** Gets the table id. */
+  UUID getId();
+
+  /** Gets the case-sensitive name of the keyspace. */
+  String getKeyspace();
```

Comment:
Maybe we could annotate return types of those methods with `@NonNull` and `@Nullable` to allow the client's code to reason about return type and guard against `null` if needed?  We did that in the java-driver and it was very useful for our clients.

Reply:
We could have the discussion outside of this PR.